genienlp/tasks/almond/__init__.py

```diff
@@ -312,4 +309,54 @@ def _make_example(self, parts):
                                 tokenize=self.tokenize, lower=False)
 
     def get_splits(self, root, **kwargs):
-        return AlmondDataset.splits(root=root, task_name='almond/agent', make_example=self._make_example, **kwargs)
\ No newline at end of file
+        return AlmondDataset.return_splits(path=os.path.join(root, 'almond/agent'), make_example=self._make_example, **kwargs)
+    
+    
+@register_task('almond_multilingual')
+class AlmondDialoguePolicy(BaseAlmondTask):
+    """Multi-Language task for Almond
+    """
+    def _is_program_field(self, field_name):
+        return field_name == 'answer'
+
+    @property
+    def metrics(self):
+        return ['em', 'bleu']
+
+    def _make_example(self, parts, language=None):
+        _id, sentence, target_code = parts
+        question = 'translate from {} to thingtalk'.format(language)
+        context = sentence
+        answer = target_code
+        return Example.from_raw(self.name + '/' + language + '/' + _id, context, question, answer,
+                                tokenize=self.tokenize, lower=False)
+    
+    def combine_datasets(self, datasets):
+        splits = defaultdict()
+        assert len(datasets) >= 1
+        fields = [field for field in datasets[0]._fields if getattr(datasets[0], field) is not None]
+        
+        for field in fields:
+            all_examples = []
+            for dataset in datasets:
+                all_examples.extend(getattr(dataset, field).examples)
+            splits[field] = AlmondDataset('', make_example=None, examples=all_examples)
```

Comment:
it doesn't need to be an AlmondDataset here, right? it could be a plain dataset

Reply:
True. I wanted to be consistent so each task uses its own dataset. I can change it though.

Reply:
As the other comment says, it's very ugly to have two completely different signatures for the same function, without sharing any of the code.